atomate/vasp/firetasks/parse_outputs.py

```diff
@@ -775,6 +776,590 @@ def run_task(self, fw_spec):
+            rkey = ""
+            if int(incar_dict["ICHARG"]) == 11:
+                use_calc = True
+                rkey = keys[1]
+            else:
+                use_calc = False
+                if relax_nonmagnetic:
+                    if int(incar_dict["ISPIN"]) == 2:
+                        use_calc = True
+                else:
+                    use_calc = True
+
+                if use_calc:
+                    is_gs = True
+                    for i in range(num_perturb_sites):
+                        v_up = float(incar_dict['LDAUU'][i])
+                        v_dn = float(incar_dict['LDAUJ'][i])
+                        if v_up != 0.0 or v_dn != 0.0:
+                            is_gs = False
+                    if is_gs:
+                        rkey = keys[0]
+                    else:
+                        rkey = keys[2]
+                else:
+                    rkey = ""
+
+            if use_calc:
+                try:
+                    # perform magnetic ordering analysis
+                    struct_final = Structure.from_dict(d["output"]["structure"])
+                    analyzer_output = CollinearMagneticStructureAnalyzer(
+                        struct_final, threshold=0.61)
+                    magnet_order = analyzer_output.ordering.value
+                    if rkey == keys[0]: # store ground state ordering
+                        magnet_order_gs = magnet_order
+
+                    for i in range(num_perturb_sites):
+                        specie = struct[i].specie
+                        ldaul = ldaul_vals[i]
+
+                        # if ldaul != -1 and rkey:
+                        orbital = inv_block_dict[str(ldaul)]
+                        perturb_dict.update(
+                            {"site"+str(i): {"specie": str(specie),
+                                             "orbital": orbital}})
+
+                        # Obtain occupancy values
+                        n_tot = float(outcar_dict['charge'][i][orbital])
+                        # FIXME: Adapt for noncollinear
+                        m_z = float((outcar_dict['magnetization'][i][orbital]))
+                        n_up = 0.5*(n_tot + m_z)
+                        n_dn = 0.5*(n_tot - m_z)
+
+                        v_up = float(incar_dict['LDAUU'][i])
+                        v_dn = float(incar_dict['LDAUJ'][i])
+
+                        response_dict[rkey]['Nup_site'+str(i)].append(n_up)
+                        response_dict[rkey]['Ndn_site'+str(i)].append(n_dn)
+                        response_dict[rkey]['Ntot_site'+str(i)].append(n_tot)
+                        response_dict[rkey]['Mz_site'+str(i)].append(m_z)
+
+                        response_dict[rkey]['Vup_site'+str(i)].append(v_up)
+                        response_dict[rkey]['Vdn_site'+str(i)].append(v_dn)
+
+                        response_dict[rkey]['magnetic order'].append(magnet_order)
+
+                except Exception as exc:
```

Comment:
This `Exception` is a little too broad; why might the above code fail?

Reply:
You're absolutely right, I'll do some tests with that `Exception` removed.